Trip planning

evansiroky opened the pull request.

evansiroky committed:
refactor(server): enhance geocode plugin

- perform geocoding in background
- only do geocode if needed
- extracting independent logic into util files

evansiroky committed:
Merge branch 'server-dev' into geocode-enhancements

evansiroky committed:
refactor(geocoding): enhance geocoding

- allow client to pass geocoded data to be saved in models to avoid
server-side geocode calls

evansiroky committed:
docs(server): add in-line code documentation for timedQueue

evansiroky committed:
refactor(server): make cascade delete actually work

evansiroky committed:
refactor(server): organize server tests + add trip plan skeleton

evansiroky committed:
feat(server): calculate trip metrics for non-transit modes

evansiroky committed:
feat(server): calculate metrics for transit trips and analysis summary

evansiroky committed:
refactor(test): get most tests passing after updating mock-data

evansiroky committed:
feat(commuter-groups): improve geocoding of commuters

- make geocoding fault tolerant and able to retry automatically
- show progress of geocoding

evansiroky committed:
feat(analysis): add client-server trip planning integration

evansiroky committed:
test(all): update tests to match latest

evansiroky committed:
chore(heroku): use environment variables instead of env file with Heroku

evansiroky committed:
chore(heroku/s3): use s3 to serve static files

evansiroky committed:
style(navbar): hide username if undefined

trevorgerhardt commented on __tests__/server/profiler.js:
```diff
@@ -0,0 +1,52 @@
+    profiler({ analysisId: analysis._id, commuters: [madeUpCommuter], site: mockSite })
+
+    // wait a little bit
+    await timeoutPromise(2000)
```
If the `profiler` runs asynchronously can we have it return a promise?

trevorgerhardt commented on __tests__/server/webservices/commuter.js:
```diff
@@ -13,9 +13,9 @@ describe('commuter', () => {
+    coordinate: {
       lat: 12,
-      lon: 34
+      lng: 34
```
We should definitely have this in the first format. Let's have everything be represented in `{lon, lat}`.

trevorgerhardt reviewed (changes requested):
Mainly the lon/lat issue: https://app.asana.com/0/138995781860804/231314340016923

evansiroky committed:
refactor(latlng): use lon instead of lng

evansiroky committed:
chore(package): update to latest mastarm

evansiroky committed:
refactor(server tests): improve a few tests

- refactor profiler to return a promise
- omit items from snapshots that are calculated after save
